### PR TITLE
Fix datetime restriction noted in UG

### LINKED ISSUE #268: Datetime restriction noted in UG is not exhaustive
It does not note that 3-digit years are not allowed and 1-digit years are allowed.

### PR DESCRIPTION
**Pull request by Dinoman44**
Fixes #268

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -459,55 +459,42 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+3. **Make prefix for deadlines `dl/`:** This is in conjunction with point 2. Our current implementation of deadline is implemented with `d/DEADLINE`. If we implement a regex to accept "d/o", this would trigger an error since it would match "d/". In the future, we will change the prefix to something like `PREFIX_DEADLINE = new Prefix("dl/")`.<br>
+4. **Allow users to clear current sorting scheme:** Currently, users cannot 'undo' a sort command. In the future, we plan to add an undo and redo feature, which will complete this enhancement as well. Do refer to [the proposed undo and redo feature](#proposed-undoredo-feature).<br>
+4. **Adjust sorting function to maintain order for identical names/deadlines:** Currently, when multiple entries share the same name or deadline, sorting them may cause reordering if a different field (i.e project status) is changed afterward. The reason for this is due to a bug with the way SortedList is implemented. For details on this bug, and how we plan to enhance it in the future, refer to [Appendix: Sorted List Bug](#appendix-sorted-list-bug).<br>
```
Comment (JunLongling) [inline docs/DeveloperGuide.md:466]: should be point 5 instead of 4

**Review by JunLongling — COMMENTED** [review]
LGTM, just some numbering error

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (JunLongling) [inline docs/DeveloperGuide.md:466]:
```suggestion
5. **Adjust sorting function to maintain order for identical names/deadlines:** Currently, when multiple entries share the same name or deadline, sorting them may cause reordering if a different field (i.e project status) is changed afterward. The reason for this is due to a bug with the way SortedList is implemented. For details on this bug, and how we plan to enhance it in the future, refer to [Appendix: Sorted List Bug](#appendix-sorted-list-bug).<br>
```

Comment (Dinoman44) [reply docs/DeveloperGuide.md:466]: Good catch

**Pull request merged**